### ISSUE TITLE
Welcome Wizard for permissions shows up very frequently

### ISSUE THREAD

**Issue by Modelmat** [syncthing/syncthing-android#1669]
# What happened

The welcome wizard pops up and asks for permissions when opening the app. This happens every time after the app has been closed, but sometimes even when running in background.

# What you expected to happen

If app permissions exist, don't show the welcome screen.

# Steps to reproduce

Easiest way is just to restart the app and open it again.

n.b. OPPO's ColorOS has some aggressive power "saving" features which may be resetting the battery permissions? I'm not sure, I have manually enabled the "lock" feature (which keeps apps running in background) and turned off "background freeze" and "abonormal apps optimisation" within the battery settings.

### Version Information

    App Version: 1.16
    Syncthing Version: 1.16
    Android Version: Android 7.1.1, OPPO ColorOS V3.1

**Comment by srett** [comment]
> n.b. OPPO's ColorOS has some aggressive power "saving" features which may be resetting the battery permissions?

Yes, I have observed that ColorOS seems to reset the battery saving settings for Syncthing at least every time the App gets updated. Didn't find any way to make it permanent yet. However, I don't experience the issue of the whole wizard showing up again, just the occasional "Do you want to disable power saving for Syncthing" popup.

**Comment by Modelmat** [comment]
If I tap the notification it seems to open the app fine (always), but opening it from the app launcher seems to prompt the intro screen (sometimes). In fact opening it from the notification will work, and then the launcher can not.
